Added IaC example page documentation (#1610)
liqotech/liqo: docs/examples/provision-with-terraform.md

The Liqo webhook will add the toleration for this taint to the pods created in the liqo-enabled namespaces.
```

Since no further configuration is provided, Liqo will add a suffix to the namespace name to make it unique on the remote cluster (see the dedicated [usage page](/usage/namespace-offloading.md) for additional information concerning namespace offloading configurations).

You can now test the infrastructure you have just created by deploying an application.
For this, you can follow the [proper example section](ExamplesStartHelloWorldApplication) in the Quick Start page.

## Tear down the infrastructure

To tear down all the infrastrucure you only need to run the following command:

```bash
terraform destroy
```

This command will destroy all the resources starting from the last one created, from bottom to top.

If you want to destroy a specific resource (for example to unpeer a cluster or to unoffload a namespace) you can leverage on the `-target` flag of `destroy` command.
For example, you can run the following command to unpeer two clusters:

```bash
terraform destroy -target="liqo_peer.peer"
```

```{warning}

> The Terraform `destroy` command will destroy all resources that have a dependence on the one that has to be destroyed.

Reviewer: Missing close warning box

liqo-metric-agent-5667b979c7-snmdg         1/1     Running   0          2m8s
liqo-network-manager-5b5cdcfcf7-scvd9      1/1     Running   0          2m8s
liqo-proxy-6674dd7bbd-kr2ls                1/1     Running   0          2m8s
liqo-route-7wsrx                           1/1     Running   0          2m8s
liqo-route-sz75m                           1/1     Running   0          2m8s
liqo-webhook-7bb6dbf88d-2hvwx              1/1     Running   0          2m8s
```

### Extract auth parameters for peering

Once the Liqo installation in the remote cluster is completed, Terraform will extract the authentication parameters required to peer the local (i.e., *Rome*) cluster with the remote one (i.e., *Milan*).

This is achieved with the `liqo_generate` resource of the `liqo` Terraform provider instance, configured with either the config file or the full list of parameters of the remote cluster:

```terraform
provider "liqo" {
  alias = "milan"
  kubernetes = {
    config_path = kind_cluster.milan.kubeconfig_path
  }
}

resource "liqo_generate" "generate" {

  depends_on = [
    helm_release.install_liqo_milan
  ]

  provider = liqo.milan

}
```

> ### Run peering procedure

Reviewer 2: ```suggestion
### Run the peering procedure
```

```text
NAME    TYPE        OUTGOING PEERING   INCOMING PEERING   NETWORKING    AUTHENTICATION   AGE
milan   OutOfBand   Established        None               Established   Established      12s
```

At the same time, you should see a virtual node (`liqo-milan`) in addition to your physical nodes:

```bash
kubectl get nodes
```

```text
NAME                 STATUS   ROLES                  AGE     VERSION
liqo-milan           Ready    agent                  14s     v1.23.6
rome-control-plane   Ready    control-plane,master   7m56s   v1.23.6
rome-worker          Ready    <none>                 7m25s   v1.23.6
```

### Offload a namespace

If you want to deploy an application that is scheduled on a Liqo virtual node (hence, it is offloaded on a remote cluster), you should first create a namespace where your pod will be started.

This can be achieved with `kubernetes_namespace` of [kubernetes](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs) Terraform provider, configured with either the config file or the full list of parameters of the local cluster.

Then tell Liqo to make this namespace eligible for the pod offloading.

> 

Reviewer 2: I would personally drop this empty line (to join the two sentences, which are related).

```terraform
resource "liqo_offload" "offload" {

  depends_on = [
    helm_release.install_liqo_rome,
    kubernetes_namespace.namespace
  ]

  provider = liqo.rome

  namespace = "liqo-demo"

}
```

```{admonition} Note
Liqo virtual nodes have a [taint](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/) that prevents the pods from being scheduled on them.
The Liqo webhook will add the toleration for this taint to the pods created in the liqo-enabled namespaces.
```

Since no further configuration is provided, Liqo will add a suffix to the namespace name to make it unique on the remote cluster (see the dedicated [usage page](/usage/namespace-offloading.md) for additional information concerning namespace offloading configurations).

You can now test the infrastructure you have just created by deploying an application.
For this, you can follow the [proper example section](StartHelloWorldApplication) in the Quick Start page.

## Tear down the infrastructure

To tear down all the infrastrucure you only need to run the following command:

```bash
terraform destroy
```

> This command will destroy all the resource starting from the last one created, from bottom to top.

Reviewer 2: ```suggestion
This command will destroy all the resources starting from the last one created, from bottom to top.
```

> If you want to destroy a specific resource (for example to un-peer or to un-offload) you can leverage on the `-target` flag of `destroy` command.

Reviewer 2: ```suggestion
If you want to destroy a specific resource (for example to unpeer a cluster or to unoffload a namespace) you can leverage on the `-target` flag of `destroy` command.
```

Just for consistency, as in the rest of the documentation they are written without an hyphen.

> For example, you can run the following command to un-peer two clusters:

Reviewer 2: ```suggestion
For example, you can run the following command to unpeer two clusters:
```

```bash
terraform destroy -target="liqo_peering.peering"
```

```{warning}

> The Terraform `destroy` command will destroy all resources that have a dependence to the one that has to be destroyed.

Reviewer 2: ```suggestion
The Terraform `destroy` command will destroy all resources that have a dependence on the one that has to be destroyed.
```
